lib/jxl/modular/transform/palette.h

```diff
@@ -83,7 +83,7 @@ static pixel_type GetPaletteValue(const pixel_type *const palette, int index,
     static constexpr int kMultiplier[] = {-1, 1};
     pixel_type result =
         kDeltaPalette[((index + 1) >> 1)][c] * kMultiplier[index & 1];
-    if (bit_depth > 8) {
+    if (bit_depth > 8 && bit_depth <= 24) {
```

Comment:
What if instead we replace the bit_depth value with std::max(bit_depth, 24)?

Reply:
Yes, that would also work (std::min though).

Another, maybe more radical option would be to just remove the bitdepth scaling altogether in the default deltas. I don't really see why it is useful in the first place. It basically causes high bit depth delta palette to just boil down to 8-bit delta palette — unless of course you add custom deltas — but that doesn't really seem to be the most desirable thing to do. If you want to use delta palette on a 10-bit image, then maybe you actually want to have default deltas with smaller steps, so slow gradients can be done more accurately.

For the default absolute colors (the 4x4x4 cube and the 5x5x5 cube), bitdepth scaling does of course make sense. But for the default deltas, I think it actually doesn't really make sense.

@jyrkialakuijala what's your take on this? So what I'm proposing is to just remove the bitdepth scaling completely from the default delta definition, i.e. the current line `if (bitdepth > 8) value <<= bitdepth - 8;` in the spec. By removing that line, you get default deltas that get increasingly more fine-grained as the bitdepth increases, and at some point you'll probably need to add non-default deltas to make effective use of delta palette, but I think that makes more sense than scaling the defaults so it's always as if things are 8-bit.

Reply:
OK, in that case, shall we change it into this?
```
if (bitdepth > 8) value <<= std::min(bitdepth,24) - 8;
```

So keep the scaling as it is now for 9..24-bit, and for higher bitdepth, clamp the scaling to what is done for 24-bit (as opposed to not allowing the (delta) palette transform at all).

Reply:
The above is how it currently is in the CD of 18181-1-Ed2.

Reply:
Perfect